pyodide_build/pypabuild.py

```diff
@@ -153,6 +153,7 @@ def _build_in_isolated_env(
 
         # first install the build dependencies
         symlink_unisolated_packages(env)
+        os.environ["PIP_CONSTRAINT"] = build_env["PIP_CONSTRAINT"]
```

Comment:
Let's add a comment that this is only for in-tree build for future readers.

For OOT build, a user needs to set PIP_CONSTRAINT if needed.